Sources/NIOCore/Channel.swift

```diff
@@ -154,6 +161,28 @@ extension Channel {
     }
 }
 
+#if compiler(>=5.5.2) && canImport(_Concurrency)
+extension Channel {
+    /// Default implementation: construct a stream with the given API surface, not modifying the configuration.
+    ///
+    /// Almost all Channels will use the default implementation. A few Channels may require specialized behaviour:
+    /// those Channels can override and modify this implementation as they need.
+    @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+    @inlinable
+    public func makeAsyncStream<InboundIn>(of type: InboundIn.Type, config: NIOInboundChannelStreamConfig) async throws -> NIOInboundChannelStream<InboundIn> {
```

Comment:
What removes these handlers once the stream is no longer in use? 

If a user calls...

```swift
for _ in 1...1000 {
    _ = try await channel.makeAsyncStream(...)
}
```

...won't this suffer from the lapsed listener problem? I.e. won't we have 1000 handlers that no one cares about any more?

As we've discussed before, one ugly solution to this was to wrap the `AsyncSequence` returned in a reference type and try and use its lifecycle to deregister the listener, which kind of worked but wasn't great.

Have I missed something here, maybe lapsed listeners are reaped somehow in this proposal that I overlooked?

Reply:
Nope, you haven't missed anything here: for now we'd need to use `deinit` on the `AsyncSequence`. It's in my list of things to tackle before we merge this.